core/dbt/parser/schemas.py

```diff
@@ -351,6 +352,23 @@ def create_test_node(
         column_name: Optional[str],
     ) -> ParsedSchemaTestNode:
 
+        HASH_LENGTH = 10
+
+        def get_hashable_md(
+            data: Union[str, int, float, List, Dict]
+        ) -> Union[str, List, Dict]:
+            if type(data) == dict:
+                return {k: get_hashable_md(data[k]) for k in sorted(data.keys())}  # type: ignore
+            elif type(data) == list:
+                return [get_hashable_md(val) for val in data]  # type: ignore
+            else:
+                return str(data)
+
+        hashable_metadata = repr(get_hashable_md(test_metadata))
+        hash_list = filter(None, [name, hashable_metadata])
+        hash_string = ''.join(hash_list).encode('utf-8')
+        test_hash = md5(hash_string).hexdigest()[-HASH_LENGTH:]
```

Comment:
It looks like we could use `hexdigest(length)` and have the library choose the value instead of pulling from the end manually.

Reply:
That's only for hashers that support variable length hashes like BLAKE2. 
```
>>> hashlib.md5(b"foobar").hexdigest(5)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: HASH.hexdigest() takes no arguments (1 given)
```